### PR TITLE
Add command-line arguments `--old` and `--new`

### LINKED ISSUE #1: Command line help and argument checking
* Add a help message (showing usage and arguments) when `cleverdiff --help` is run.
* Print only a helpful error message and a non-zero return code if no arguments are given.

### LINKED ISSUE #7: Add command-line arguments `--old` and `--new`
At present input files are specified using the form:
```
cleverdiff old1.ext=new1.ext old2.ext=new2.ext ...
```
This example will compare the diff between old1.ext vs old2.ext

Change this so that cleverdiff will accept a new form of arguments which will behave identically to above by listing the old and new files separately using new options `--old` and `--new` e.g. 
```
cleverdiff --old old1.ext old2.ext --new new1.ext new2.ext
```

This issue is related to #1.

### PR DESCRIPTION
**Pull request by vhsw**
Fixes #1, fixes #7

### CODE REVIEW

File: cleverdiff/cleverdiff.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 from __future__ import absolute_import, division, print_function  # noqa
 
-import sys
-import os.path
+import os
```
Comment (andrewcbennett) [inline cleverdiff/cleverdiff.py]: This import is not necessary and can be removed.

File: cleverdiff/cleverdiff.py [hunk 2]
```diff
@@ -38,13 +38,13 @@ def summarise_results(diffseen, diffresult):
     for seenindex, seenitem in enumerate(diffseen):
         result += "--------------------------------------------\n"
         result += f"DIFF {seenindex:3d}:\n" f"{str(seenitem)}"
-        result2 = {key: [] for key in MODE_DESCRIPTION.keys()}
+        result2 = {key: [] for key in MODE_DESCRIPTION}
```
Comment (andrewcbennett) [inline cleverdiff/cleverdiff.py]: Thanks, this makes the code a bit simpler and more readable. However, as this is unrelated to the command-line argument changes, I'd prefer it to be in a separate pull request, along with the related change on line 47.

File: cleverdiff/cleverdiff.py [hunk 3]
```diff
@@ -38,13 +38,13 @@ def summarise_results(diffseen, diffresult):
...
         for hunk2, controlindex, mode in diffresult:
             if controlindex != seenindex:
                 continue
 
             # TODO: stick this assert in a test!
-            assert mode < 2, "unexpected value for mode"
+            assert mode in MODE_DESCRIPTION, "unexpected value for mode"
```
Comment (andrewcbennett) [inline cleverdiff/cleverdiff.py]: See my comment on line 41.

File: cleverdiff/cleverdiff.py [hunk 4]
```diff
@@ -85,14 +85,37 @@ def main_diff(filepairs, context_cls=None):
...
+        dest="new_files",
+        metavar="file",
+        type=str,
+        nargs="+",
+        help="list of modified files",
+    )
+
+    args = parser.parse_args()
+    if args.old_files is None or args.new_files is None:
+        raise ValueError("No files specified")
```
Comment (andrewcbennett) [inline cleverdiff/cleverdiff.py]: This exception is raised if neither `--old` nor `--new` is provided, but also if the user forgets one of them. For the latter case, could you provide a specific error for when one of them is specified but not the other (e.g. "both --old and --new arguments must be specified")?

**Comment by vhsw** [comment]
I've added some tests and reenabled original behavior

File: cleverdiff/cleverdiff.py (hunk 4 above)
Comment (andrewcbennett) [reply cleverdiff/cleverdiff.py]: Would still like a specific error to be shown for when `--old` or `--new` is specified but not both (e.g. "both --old and --new arguments must be specified").

Comment (vhsw) [reply cleverdiff/cleverdiff.py]: The exception is added

**Comment by andrewcbennett** [comment]
Thank you kindly for your contribution @vhsw - have a 🥇 for being the first contributor! 😄
